### PR TITLE
Viewer for Game History

### PR DESCRIPTION
**Pull request by zouroboros**
Viewer for the game history files. For history files see #66.

### CODE REVIEW

**Comment by zouroboros** [comment]
I just opened this for discussion, please don't merge it at this point.

**Comment by zouroboros** [comment]
Now it is ready to be merged. At least from my side-

**Comment by s-will** [comment]
Hi @zouroboros, having the possibility to store and view the history is a very good idea:) Thanks for contributing this!

File: A6/6nimmt.py [hunk 1]
```diff
@@ -276,11 +284,10 @@ def play_game(cards, players, number_of_cards_per_player, number_of_rows, time_l
...
+
     name_players(players)
-    
+
     for i in range(args.games):
         cards = list( range(1,args.cards+1) )
         random.shuffle(cards)
-        game_statistitcs = play_game(cards, players, args.number_of_cards_per_player, args.number_of_rows, args.time_limit)
-    
+        game_statistitcs = play_game(cards, players, args.number_of_cards_per_player, args.number_of_rows, args.time_limit, args.history_file + "_game=" + str(i))
```
Comment (s-will) [inline A6/6nimmt.py]: This line fails if history file is not given. However, history file should be optional.

File: A6/6nimmt.py [hunk 2]
```diff
@@ -130,74 +131,76 @@ def difference(x,y):
...
     game_status = initial_game_status(cards, players, number_of_cards_per_player, number_of_rows, time_limit)
-    
+
     logging.debug("Inititial status: "+str(game_status))
 
 
     for round_index in range(number_of_cards_per_player):
         logging.info("-------------------- Play round "+str(round_index+1)+"--------------------")
         logging.info("Game status: "+str(game_status))
+        history.append(copy.deepcopy(game_status.copy()))
```
Comment (s-will) [inline A6/6nimmt.py]: Shouldn't the deepcopy be sufficient---why do we need the copy() in game_status.copy()?

Comment (zouroboros) [inline A6/6nimmt.py]: Yes it is sufficient. I changed it.

File: A6/6nimmt.py (hunk 1 above)
Comment (zouroboros) [inline A6/6nimmt.py]: Definitely. I fixed it.

**Comment by zouroboros** [comment]
I fixed the two point you mentioned and changed the game master so that all games are write to one file. (I also updated the viewer to be compatibly with the new history file format)

**Comment by s-will** [comment]
Many thanks!

**Pull request merged**